open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataTopicConnector.java

```diff
@@ -347,4 +347,14 @@ public String getPrintableProperties(Properties   liveProperties)
 
         return printableProperties;
     }
+    
+    /**
+     * Gets the number of events that have not been processed yet.
+     * 
+     * @return
+     */
+    public int getNumberOfUnprocessedEvents() {
```

Comment:
I am not very familiar with the egeria code. But I glanced at it. It looks like, events from this list is copied to another list and all events from this list is removed. That does not necessarily mean that events were processed. So there are still large number of un porcessed events while we poll again.

Reply:
@guptaneeru Good point.  For the fix to be effective, the property that limits the number events returned by polling also needs to be used.  It was this list that was seen growing unchecked in the heap dump, to the fix I've added is targetted at limiting its size.  I did not see the copy of the list being processed in the profile.

Reply:
Right, my point was we may shift the problem from one place to another. But again, some body more familiar with egeria code should evaluate that.

Reply:
@guptaneeru I don't think we are.  Basically, with this change we are limiting the number of kafka message that can be in memory.  The change makes the value of `max.poll.records` more important.  In the worst case, if there is only one consumer and max.pool.records > 100 (which will be most of the time), there will be at most 2*max.poll.records in memory -- one set that is being processed in the other thread and one in queue in this class.  If max.poll.records < 100, then there will be at most 200 in memory -- 100 being processed and 100 in the queue.  There is no way we would end up with a situation where there are 50,000 events in memory.